test/common/core/corediff.scenarios.spec.js

```diff
@@ -879,4 +879,98 @@ describe('corediff scenarios', function () {
             })
             .nodeify(done);
     });
+
+    // Multiple moving
+    it('should be able to merge multiple not conflicting but cascading moves', function (done) {
+        var originRoot,
+            paths = [];
+
+        loadRootAndFCO(context.rootHash)
+            .then(function (r) {
+
+                for (var i = 0; i < 4; i += 1) {
+                    paths.push(core.getPath(core.createNode({parent: r.root, base: r.fco, relid: 'r' + i})));
+                }
+                originRoot = r.root;
+
+                return save(r.root);
+            })
+            .then(function (rootHash) {
+                return Q.all([
+                    loadRootAndFCO(rootHash, paths),
+                    loadRootAndFCO(rootHash, paths),
+                ]);
+            })
+            .then(function (trees) {
+
+                //move 0 under 1
+                core.moveNode(trees[0][paths[0]], trees[0][paths[1]]);
+                //move 1 under 2
+                core.moveNode(trees[1][paths[1]], trees[1][paths[2]]);
+
+                // Save to ensure the added nodes are persisted.
+                return Q.all([
+                    save(trees[0].root),
+                    save(trees[1].root)
+                ])
+                    .then(function () {
+                        return Q.all([
+                            core.generateTreeDiff(originRoot, trees[0].root),
+                            core.generateTreeDiff(originRoot, trees[1].root),
+                        ]);
+                    });
+            })
+            .then(function (diffs) {
+                var merge = core.tryToConcatChanges(diffs[0], diffs[1]);
+
+                expect(merge.items).to.have.length(0);
```

Comment:
It's nice and all that there's no conflicts, but it would be nice to check that the resulting model has the /2/1/0 structure, same goes for the test below..